Bug 1130525: Add ReportMsg for S3SplitFile plugins

mreid-moz opened the pull request:
Implement the ReportMsg interface to provide visibility into
errors and progress in S3SplitFileInput and S3SplitFileOutput.

mreid-moz committed:
Bug 1130525: Add ReportMsg for S3SplitFile plugins

Implement the ReportMsg interface to provide visibility into
errors and progress in S3SplitFileInput and S3SplitFileOutput.

trink commented on heka/plugins/s3splitfile/s3splitfile_input.go:
```diff
@@ -10,19 +10,26 @@ import (
+	schema                 Schema
+	stop                   chan bool
+	listChan               chan string
+	processFileCount       int64
```
To avoid any memory alignment issues on 32 bit systems place these int64 values at the beginning of the structure

trink commented on heka/plugins/s3splitfile/s3splitfile_input.go:
```diff
@@ -204,6 +212,16 @@ func (input *S3SplitFileInput) fetcher(runner pipeline.InputRunner, wg *sync.Wai
+	message.NewInt64Field(msg, "ProcessFileFailures", atomic.LoadInt64(&input.processFileFailures), "count")
+	message.NewInt64Field(msg, "ProcessMessageCount", atomic.LoadInt64(&input.processMessageCount), "count")
+	message.NewInt64Field(msg, "ProcessMessageFailures", atomic.LoadInt64(&input.processMessageFailures), "count")
+	message.NewInt64Field(msg, "ProcessMessageSumSize", atomic.LoadInt64(&input.processMessageSumSize), "B")
```
The sandbox input uses ProcessMessageBytes.  We should select a standardized name.

trink commented on heka/plugins/s3splitfile/s3splitfile_output.go:
```diff
@@ -11,26 +11,36 @@ import (
+	bucket                     *s3.Bucket
+	publishChan                chan PublishAttempt
+	shuttingDown               bool
+	processFileCount           int64
```
Same, move these to the beginning of the struct

mreid-moz committed:
Move report fields, tweak naming.

Move the int64 fields to the beginning of the struct to ensure
proper alignment on 32bit systems, and use consistent naming for
the "Size" fields.

mreid-moz committed:
Add tracking of remaining data.

Extra bytes left at the end of the file indicate data corruption,
either during the file read, or worse, in the source data in S3.

trink commented on heka/plugins/s3splitfile/s3splitfile_input.go:
```diff
@@ -204,6 +215,17 @@ func (input *S3SplitFileInput) fetcher(runner pipeline.InputRunner, wg *sync.Wai
+	message.NewInt64Field(msg, "ProcessMessageCount", atomic.LoadInt64(&input.processMessageCount), "count")
+	message.NewInt64Field(msg, "ProcessMessageFailures", atomic.LoadInt64(&input.processMessageFailures), "count")
+	message.NewInt64Field(msg, "ProcessMessageBytes", atomic.LoadInt64(&input.processMessageBytes), "B")
+	message.NewInt64Field(msg, "RemainingDataBytes", atomic.LoadInt64(&input.remainingDataBytes), "B")
```
From an analysis perspective it is really dropped/discarded/corrupt bytes and it would still fall under your ProcessFile\* grouping

mreid-moz replied:
How about `ProcessFileLeftoverBytes`?

trink replied:
I would prefer DiscardedBytes (since they were read and thrown away without any processing)

mreid-moz committed:
Rename RemainingDataBytes

Rename RemainingDataBytes to ProcessFileDiscardedBytes.
